bbq/compiler/compiler.go

```diff
@@ -52,8 +52,13 @@ type Compiler[E, T any] struct {
 	usedImportedGlobals []*global
 	controlFlows        []controlFlow
 	currentControlFlow  *controlFlow
+	returns             []returns
+	currentReturn       *returns
 	staticTypes         []T
 
+	postConditionsIndices map[ast.Declaration]int
+	postConditionsIndex   int
```

Comment:
Could you please add a comment explaining the purpose of these new fields? Is it a "statement index"?

Reply:
I had added the comment in the desugar, where this was used. Copied the same here: https://github.com/onflow/cadence/pull/3816/commits/e0c6f34241118d3c50ff1df30783b32beaaaff5c